pkg/storage/mvcc_test.go

```diff
@@ -5248,6 +5248,58 @@ func TestResolveIntentWithLowerEpoch(t *testing.T) {
 	}
 }
 
+// TestTimeSeriesMVCCStats ensures that merge operations
+// result in expected increase in timeseries data
+func TestTimeSeriesMVCCStats(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	for _, engineImpl := range mvccEngineImpls {
+		t.Run(engineImpl.name, func(t *testing.T) {
+			engine := engineImpl.create()
+			defer engine.Close()
+			var ms = enginepb.MVCCStats{}
+
+			// Perform the same sequence of merges on the same key
+			// twice and record timeseries data
+			var firstSysBytes = ms.SysBytes
+			var firstLiveBytes = ms.LiveBytes
+			var firstValBytes = ms.ValBytes
+			if err := MVCCMerge(ctx, engine, &ms, testKey1, hlc.Timestamp{Logical: 1}, tsvalue1); err != nil {
+				t.Fatal(err)
+			}
+
+			var secondSysBytes = ms.SysBytes
+			var secondLiveBytes = ms.LiveBytes
+			var secondValBytes = ms.ValBytes
+			if err := MVCCMerge(ctx, engine, &ms, testKey1, hlc.Timestamp{Logical: 1}, tsvalue1); err != nil {
+				t.Fatal(err)
+			}
+
+			// Ensure timeseries metrics increases as expected
+			if isSysLocal(testKey1) {
+				var expectedSysBytes = firstSysBytes + int64(len(tsvalue1.RawBytes))
+
+				if secondSysBytes != expectedSysBytes {
+					t.Fatalf("second merged SysBytes value %v differed from expected SysBytes value %v", secondSysBytes, expectedSysBytes)
+				}
+			} else {
+				var expectedLiveBytes = firstLiveBytes + int64(len(tsvalue1.RawBytes))
+				var expectedValBytes = firstValBytes + int64(len(tsvalue1.RawBytes))
```

Comment:
I'd be curious to see if this failed previously. I expect it to given our constant addition each time.

Reply:
I'll try running this new test with the previous code to confirm!

Reply:
Confirming that the test does fail without the change: 
Before: 
```
=== RUN   TestTimeSeriesMVCCStats
=== RUN   TestTimeSeriesMVCCStats/rocksdb
=== RUN   TestTimeSeriesMVCCStats/pebble
--- FAIL: TestTimeSeriesMVCCStats (0.01s)
    --- FAIL: TestTimeSeriesMVCCStats/rocksdb (0.01s)
        mvcc_test.go:5282: second merged LiveBytes value 92 differed from expected LiveBytes value 80
    --- FAIL: TestTimeSeriesMVCCStats/pebble (0.00s)
        mvcc_test.go:5282: second merged LiveBytes value 92 differed from expected LiveBytes value 80
FAIL
FAIL	github.com/cockroachdb/cockroach/pkg/storage	1.203s
FAIL
make: *** [test] Error 1
```
After: 
```
=== RUN   TestTimeSeriesMVCCStats
=== RUN   TestTimeSeriesMVCCStats/rocksdb
=== RUN   TestTimeSeriesMVCCStats/pebble
--- PASS: TestTimeSeriesMVCCStats (0.01s)
    --- PASS: TestTimeSeriesMVCCStats/rocksdb (0.01s)
    --- PASS: TestTimeSeriesMVCCStats/pebble (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/pkg/storage	1.076s
```